api/v1beta1/azurecluster_default.go

```diff
@@ -119,7 +119,7 @@ func (c *AzureCluster) setSubnetDefaults() {
 		if subnet.SecurityGroup.Name == "" {
 			subnet.SecurityGroup.Name = generateNodeSecurityGroupName(c.ObjectMeta.Name)
 		}
-		cpSubnet.SecurityGroup.SecurityGroupClass.setDefaults()
+		subnet.SecurityGroup.SecurityGroupClass.setDefaults()
```

Comment:
I'm not sure that is a bug or by design, do the defaults apply to node subnets as well

https://capz.sigs.k8s.io/topics/custom-vnet#custom-security-rules for reference

Reply:
Also, is this a recent regression or it's always been like this?

Reply:
@CecileRobertMichon It looks like a regression when we transitioned from release-1.1 to release-1.2. Quite an old one I guess 

release-1.1: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/4a53842ec20f3a9fcdfcbb6605d5a32c4737ea49/api/v1beta1/azurecluster_default.go#L121

release-1.2: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/357e0b7a97a03f2481cf85bf34cfb8ea13747913/api/v1beta1/azurecluster_default.go#L120

I was actually working on providing support for a single subnet for both cp and node, hence was going though the code and found it weird that we are updating the cpSubnet security group rules in the for loop when it was already updated over [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6987809212f86a583c7ab0b4fc47a7a0be8a5c3e/api/v1beta1/azurecluster_default.go#L97)

Hence felt it was a bug, confirmed it with @mboersma on slack and then raised a PR :smile: